test/metric_dimension/aggregation_dimensions_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package metric_dimension
+
+import (
+	"log"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+type AggregationDimensionsTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+var _ test_runner.ITestRunner = (*AggregationDimensionsTestRunner)(nil)
+
+func (t *AggregationDimensionsTestRunner) Validate() status.TestGroupResult {
+	mm := t.GetMeasuredMetrics()
+	r := make([]status.TestResult, len(mm))
+	for i, m := range mm {
+		r[i] = t.validate(m)
+	}
+	return status.TestGroupResult{
+		Name:        t.GetTestName(),
+		TestResults: r,
+	}
+}
+
+func (t *AggregationDimensionsTestRunner) GetTestName() string {
+	return "AggregationDimensionsTestRunner"
+}
+
+func (t *AggregationDimensionsTestRunner) GetAgentConfigFileName() string {
+	return "aggregation_dimensions.json"
+}
+
+func (t *AggregationDimensionsTestRunner) GetMeasuredMetrics() []string {
+	return []string{"mem_used_percent", "cpu_usage_user"}
+}
+
+// getExpectedDimensions returns a list of expected dimensions for the given metric name.
+// This is based on the JSON configuration.
+func getExpectedDimensions(metricName string) [][]string {
+	switch metricName {
+	case "mem_used_percent":
```

Comment:
We could consider remove `cpu_usage_user` since overall `mem_used_percent` is an entire population that covers all the use case for `cpu_usage_user` and the way. we publish them all the same (unless plugin such as ethtool or nvidia which does not have `append_dimensions`)